src/Turbo/src/Doctrine/BroadcastListener.php

```diff
@@ -126,28 +130,28 @@ public function reset(): void
 
     private function storeEntitiesToPublish(EntityManagerInterface $em, object $entity, string $property): void
     {
-        $class = ClassUtil::getEntityClass($entity);
+        $entityClass = $this->doctrineClassResolver->resolve($entity, $em);
 
-        if (!isset($this->broadcastedClasses[$class])) {
-            $this->broadcastedClasses[$class] = [];
-            $r = new \ReflectionClass($class);
+        if (!isset($this->broadcastedClasses[$entityClass])) {
+            $this->broadcastedClasses[$entityClass] = [];
+            $r = new \ReflectionClass($entityClass);
 
             if ($options = $r->getAttributes(Broadcast::class)) {
                 foreach ($options as $option) {
-                    $this->broadcastedClasses[$class][] = $option->newInstance()->options;
+                    $this->broadcastedClasses[$entityClass][] = $option->newInstance()->options;
                 }
             } elseif ($this->annotationReader && $options = $this->annotationReader->getClassAnnotations($r)) {
                 foreach ($options as $option) {
                     if ($option instanceof Broadcast) {
-                        $this->broadcastedClasses[$class][] = $option->options;
+                        $this->broadcastedClasses[$entityClass][] = $option->options;
                     }
                 }
             }
         }
 
-        if ($options = $this->broadcastedClasses[$class]) {
+        if ($options = $this->broadcastedClasses[$entityClass]) {
             if ('createdEntities' !== $property) {
-                $id = $em->getClassMetadata($class)->getIdentifierValues($entity);
+                $id = $this->doctrineIdAccessor->getEntityId($entity, $em);
```

Comment:
Could you illustrate precisely what cannot be done before, what it would look like, and same after your changes ? :)